src/biogeochem/CNVegNitrogenStateType.F90

```diff
@@ -1157,8 +1157,8 @@ subroutine Summary_nitrogenstate(this, bounds, num_allc, filter_allc, &
          this%totn_patch(bounds%begp:bounds%endp), &
          this%totn_p2c_col(bounds%begc:bounds%endc))
 
-    do fc = 1,num_allc
-       c = filter_allc(fc)
+    do fc = 1,num_soilc
+       c = filter_soilc(fc)
```

Comment:
@billsacks suggested initializing the variable that was giving nans to zero before call p2c. I had tried this last week (at least for the CNVegCarbonStateType error) and failed for some reason, so for now I will keep the do-loop changes. The izumi test-suite suggests that nothing breaks due the change to the do-loops